happiness-backend/api/dao/happiness_dao.py

```diff
@@ -64,7 +66,74 @@ def get_happiness_by_group_count(user_ids, page, n):
         Happiness.timestamp.desc()) \
         .paginate(page=page, per_page=n, error_out=False)
 
+def get_happiness_by_filter(user_id, page, per_page, start, end, low, high, query):
+    """
+    Filters according to the provided arguments. Checks to see what filters to apply. Will not apply the following
+    filters with the following arguments
+     low/high = 11
+     start/end = None
+     text = "".
+    """
+    if start is not None and low != 11 and query != "":
+        return db.paginate(
+            select=
+            (select(Happiness).where(Happiness.timestamp.between(start, end),
+                                     Happiness.user_id == user_id).where(
+                    Happiness.value >= low, Happiness.value <= high,
+                    Happiness.user_id ==
+                    user_id).where(Happiness.comment.like(f"%{query}%")).order_by(Happiness.timestamp.asc())),
+            per_page=per_page,
+            page=page
+        )
+    elif start is not None and low != 11:
```

Comment:
You can definitely rewrite this method to not address each case in a `elif` chain. You could probably do a select statement, then apply `.where` whenever filters are present. For example you start by doing select happiness, then if low and high are there you `.where` on that, then check the next filter and apply `.where` again to the same query object, etc..

Reply:
I agree :-)

Reply:
i think i understand this point but i don't really undestand exactly what you mean or how to do this. correct me if im wrong, but from my current understanding, instead of putting my boolean expressions which are checking to see which arguments are "real" on the outside of the select statement, i should instead put this logic inside the select statement, like this
 select=
        (select(Happiness).where( (if start is not None : 
            
            (Happiness.timestamp.between(start, end)) & if low != 11: 
Happiness.value >= low, Happiness.value <= high)
.order_by(Happiness.timestamp.asc())),
        per_page=per_page,
        page=page
    )
or something like this, ignore my bad indenting sorry, but i don't really know how to format this bc where is expecting filters, not boolean logic to my knowledge, and if we put the boolean logic outside of the where statement, or outside of the select statement, then aren't we back to the original logic i had, needing to have multiple select statements?

Reply:
i think i understand this point but i don't really undestand exactly what you mean or how to do this. correct me if im wrong, but from my current understanding, instead of putting my boolean expressions which are checking to see which arguments are "real" on the outside of the select statement, i should instead put this logic inside the select statement, like this
 select=
        (select(Happiness).where( (if start is not None : 
            
            (Happiness.timestamp.between(start, end)) & if low != 11: 
Happiness.value >= low, Happiness.value <= high)
.order_by(Happiness.timestamp.asc())),
        per_page=per_page,
        page=page
    )
or something like this, ignore my bad indenting sorry, but i don't really know how to format this bc where is expecting filters, not boolean logic to my knowledge, and if we put the boolean logic outside of the where statement, or outside of the select statement, then aren't we back to the original logic i had, needing to have multiple select statements?